frontend/src/lib/uPlotLib/utils/getSeriesData.ts

```diff
@@ -23,20 +26,31 @@ const paths = (
 	return renderer(u, seriesIdx, idx0, idx1, extendGap, buildClip);
 };
 
-const getSeries = (
-	apiResponse?: MetricRangePayloadProps,
-	widgetMetaData: QueryData[] = [],
-	graphsVisibilityStates?: boolean[],
-): uPlot.Options['series'] => {
+const getSeries = ({
+	apiResponse,
+	widgetMetaData,
+	graphsVisibilityStates,
+	panelType,
+}: GetSeriesProps): uPlot.Options['series'] => {
 	const configurations: uPlot.Series[] = [
 		{ label: 'Timestamp', stroke: 'purple' },
 	];
 
 	const seriesList = apiResponse?.data.result || [];
 
+	const sortedSeriesList = seriesList.sort((a, b) => {
+		const avgA =
+			a.values.reduce((acc, curr) => acc + parseFloat(curr[1]), 0) /
+			a.values.length;
+		const avgB =
+			b.values.reduce((acc, curr) => acc + parseFloat(curr[1]), 0) /
```

Comment:
What are the possibilities of length being 0?

Reply:
Added the safety check